server/src/main/java/org/opensearch/index/codec/customcodecs/Lucene95CustomCodec.java

```diff
@@ -8,19 +8,27 @@
 
 package org.opensearch.index.codec.customcodecs;
 
-import org.apache.lucene.codecs.StoredFieldsFormat;
+import org.apache.logging.log4j.Logger;
 import org.apache.lucene.codecs.FilterCodec;
+import org.apache.lucene.codecs.StoredFieldsFormat;
 import org.apache.lucene.codecs.lucene95.Lucene95Codec;
+import org.opensearch.index.codec.PerFieldMappingPostingFormatCodec;
+import org.opensearch.index.mapper.MapperService;
 
-import java.util.Locale;
-
-abstract class Lucene95CustomCodec extends FilterCodec {
+/**
+ *
+ * Extends {@link FilterCodec} to reuse the functionality of Lucene Codec.
+ * Supports two modes zstd and zstd_no_dict.
+ *
+ * @opensearch.internal
+ */
+public abstract class Lucene95CustomCodec extends FilterCodec {
     public static final int DEFAULT_COMPRESSION_LEVEL = 6;
```

Comment:
would this be exposed as a setting to change compression level

Reply:
Yes, this is the compression level for which the settings might be exposed. This will be taken up separately because there are caveats when we do enable dynamic settings for it. We might also need to consider how much control we want to give to the customer.

Reply:
to start with, it could be node level setting only?

Reply:
Agree, this is certainly an index level setting, along with the code itself, adding it incrementally sounds good

Reply:
Make sense to keep it as index level setting. We can pick it up incrementally.